Source/WebCore/platform/gstreamer/GStreamerElementHarness.cpp

```diff
@@ -208,43 +188,75 @@ bool GStreamerElementHarness::pushEvent(GstEvent* event)
     return result;
 }
 
-GRefPtr<GstBuffer> GStreamerElementHarness::pullBuffer()
+GStreamerElementHarness::Stream::Stream(GRefPtr<GstPad>&& pad)
+    : m_pad(WTFMove(pad))
 {
-    GST_LOG_OBJECT(m_element.get(), "%zu buffers currently queued", m_bufferQueue.size());
+    m_targetPad = gst_pad_new_from_static_template(&s_harnessSinkPadTemplate, "sink");
+    gst_pad_link(m_pad.get(), m_targetPad.get());
+
+    gst_pad_set_chain_function_full(m_targetPad.get(), reinterpret_cast<GstPadChainFunction>(+[](GstPad* pad, GstObject*, GstBuffer* buffer) -> GstFlowReturn {
+        auto& stream = *reinterpret_cast<GStreamerElementHarness::Stream*>(pad->chaindata);
+        return stream.chainBuffer(buffer);
+    }),  this, nullptr);
+    gst_pad_set_query_function_full(m_targetPad.get(), reinterpret_cast<GstPadQueryFunction>(+[](GstPad* pad, GstObject* parent, GstQuery* query) -> gboolean {
+        auto& stream = *reinterpret_cast<GStreamerElementHarness::Stream*>(pad->querydata);
+        return stream.sinkQuery(pad, parent, query);
+    }), this, nullptr);
+    gst_pad_set_event_function_full(m_targetPad.get(), reinterpret_cast<GstPadEventFunction>(+[](GstPad* pad, GstObject*, GstEvent* event) -> gboolean {
+        auto& stream = *reinterpret_cast<GStreamerElementHarness::Stream*>(pad->eventdata);
+        return stream.sinkEvent(event);
+    }), this, nullptr);
+
+    gst_pad_set_active(m_targetPad.get(), TRUE);
+}
+
+GStreamerElementHarness::Stream::~Stream()
+{
+    gst_pad_set_active(m_targetPad.get(), FALSE);
+    {
```

Comment:
This scope is not needed. There is nothing below that scope so it is going to unlock anyway.